src/Parser/TokenIterator.php

```diff
@@ -19,11 +20,16 @@ class TokenIterator
 
 	private int $index;
 
+	/** @var array<Comment> */
+	private array $comments = [];
+
 	/** @var int[] */
 	private array $savePoints = [];
 
 	/** @var list<int> */
-	private array $skippedTokenTypes = [Lexer::TOKEN_HORIZONTAL_WS];
+	private array $skippedTokenTypes = [
+		Lexer::TOKEN_HORIZONTAL_WS,
+		Lexer::TOKEN_COMMENT];
```

Comment:
Don't forget about `addEndOfLineToSkippedTokens()` and `removeEndOfLineFromSkippedTokens()`.